---
layout: post
date: 'Sat 2012-06-09 14:57:02 +0800'
slug: "on-casava"
title: "On CASAVA"
description: ""
category: BioInformatics
tags: [Galaxy_Original, illumina, CASAVA]
---
{% include JB/setup %}

官网注册帐号后就可以下载，最新的1.8.2已经支持每个样品输出一组FastQ文件。

编译安装运行都不需要root权限，以前在BIOPIC听她们说跑CASAVA必须root权限我还奇怪了些许……

贴个命令行备案：

	/opt/CASAVA/bin/configureBclToFastq.pl --input-dir /bak/seqdata/original/120523_SN483_0244_BC0M01ACXX/Data/Intensities/BaseCalls --output-dir /bak/seqdata/raw/120523_SN483_0244_BC0M01ACXX --sample-sheet /bak/seqdata/original/0244_BC0M01ACXX-SS.csv --no-eamss --with-failed-reads --fastq-cluster-count -1 --mismatches 1 --tiles s_[78] 2>do.err
	
	cd /bak/seqdata/raw/120523_SN483_0244_BC0M01ACXX
	make -j23 -l23.5 >make.txt 2>make.err &

`--fastq-cluster-count -1`就是设置只输出一组`.fq.gz`的。其实`/Data/Intensities/BaseCalls/`下的`.bcl`文件已经是只有最好碱基(2bit)与质量值(6bit)的东西了，所以，安装通常的RTA直接在Windows下出碱基来看的话，CASAVA就是转格式与按index拆分而已。那个`--mismatches 1`就是设置index上的mismatch数，默认是0。

不过`./Data/Intensities/L00[1-8]/s_*.clocs`是CASAVA需要的，只是旁边的`*.cif`和`*_T.FWHMMap`可以不要。

`make`时其实只会每个fq出一个进程，就是说，PE的FlowCell最多16进程同时跑。
由于`.bcl`文件是每Cycle一个，所以内存够大的话，缓存能把运行时间缩小5倍多（大致估的）。

然后，运行时看到：

	[configureBclToFastq.pl] WARNING: 'LocationFileType' element not found in .../Data/Intensities/BaseCalls/../RTAConfiguration.xml

网上找到 [解答](http://seqanswers.com/forums/showthread.php?t=13172 "CASAVA v1.8 (Bcl to Fastq)")：

> I actually asked tech support about this, here is their answer:

	Hi Louis,
	This warning message you see is due to a slight incompatibility between RTA/OLB versions and CASAVA 1.8 standard requirements and there should not be any issues during data analysis as it is not an error message.

	Regards,
	Sid


	-----<Email Received>-----
	From: ..
	Sent: 6/26/2011 12:18 PM PDT
	To: techsupport@illumina.com
	CC:
	Subject: configureBclToFastq.pl warning

	When running configureBclToFastq.pl I got this:

	WARNING: 'LocationFileType' element not found in
	/data/hiseq67/110617_SN909_0050_BD040RACXX_0047HS11B/Data/Intensities/BaseCalls/../RTAConfiguration.xml


	What can we do to remove this warning?
	It's not an important warning but I just want to make sure we fill out
	everything in case.

	Thanks
	Louis
